scipy/signal/ltisys.py

```diff
@@ -2494,3 +2496,218 @@ def place_poles(A, B, poles, method="YT", rtol=1e-3, maxiter=30):
     full_state_feedback.nb_iter = nb_iter
 
     return full_state_feedback
+
+
+def controller_Hessenberg(A,B, C=None, tol=None):
+    r"""
+    Reduce the linear system ``AX + BU`` to the controller Hessenberg form
+    H, B_bar where P*A*P.T = H and P*B = Bbar with P being an orthogonnal
```

Comment:
FWIW, I would avoid bars, tildes and hats as much as reasonable, both in code and maths. But it's a matter of taste anyway